### ISSUE TITLE
LM loss

### ISSUE THREAD

**Issue by wangjiyuan9** [Haochen-Wang409/ross#17]
作者你好，我正在复现您的工作。不过我是直接在您的ROSS上微调的。其实我想问下，这个release的ROSS应该不是那个和sota比较的大ROSS吧？是558k pt+787k ft的版本吧？
现在我就在558k上预训练，但是LM loss完全不下降，相比之下，VMloss 确实挺低的：

<img width="1585" alt="Image" src="https://github.com/user-attachments/assets/4d8b8f5a-150d-4b14-a416-664ce5a7b362" />

请问这是正常的吗？

**Comment by wangjiyuan9** [comment]
另外，请问您能细致的说一下载Image Net ft的细节吗？这个意思是在不添加LLM下，只训练denoiser吗？paper中说用5个epoch，但如果没有llm的训练，是否更少的epoch就可以得到好的diffusion model？

**Comment by Haochen-Wang409** [comment]
1. Released 是最终与 sota 比较的 Ross
2. 看上去不是特别正常，你是加载了 Ross 的权重然后在 558K 上训练的吗？
3. ImageNet finetune 的流程大致是：图 -> SigLIP (frozen) -> Qwen2 (frozen) -> denoiser (trainable) -> predicted noise。关于”但如果没有llm的训练，是否更少的epoch就可以得到好的diffusion model“这一点，我没有特别理解，你能详细解释一下吗？

**Comment by wangjiyuan9** [comment]
1.OK感谢，但我居然用clip训练脚本训练起来了，这个release的我记得是siglip吧？
2.是的，加载ross权重之后再558k上训练的，另外我下载了qwen 2.5b ft版本从头训练，发现lm loss从6降低到2.5左右。目前之是做了第一阶段。现在我最大的苦恼就是不理解为啥训练的lm_loss一直在2以上，无论使用ROSS训练还是qwen2.5，这很奇怪。给您我的loss曲线（qwen2.5橙色，ROSS蓝色）

<img width="1144" alt="Image" src="https://github.com/user-attachments/assets/bfc12c8a-aa2e-49d9-9847-f80855cfe3d7" />

3.额我之前的理解是您有只训练denoiser的情况（这里我们是假设image net给打了caption，直接把caption当做condition来训练），现在看您似乎依然用的Qwen来finetune的。不过问题是，这里的llm文字输入是啥呢？完全没有文字输入吗？
4.感谢您的回复，我之前一直有一个核心问题：如何保证SigLIP（冻结的）提取出来的特征能让后续的网络恢复出来原图呢？（比如SigLIP不够强大，本身就有一定的对图像的损耗，那后续学出花来也不应该能恢复图。）为何您从来没想过解冻CLIP或者SigLIP呢？

**Comment by Haochen-Wang409** [comment]
1. release 的是 SigLIP
2. 第一阶段没有打开 LLM，loss 在 2 以上属于正常水平。SFT 阶段会下降到 1 以下
3. LLM 没有文字输入，只是输入图片。假设图片被 CLIP encode 为 576 token，那么 LLM 的输入、输出都会有 576 token。我们将输出的 576 token 作为 denoiser 的 condition
4. 其实解冻 SigLIP 可能甚至与 Ross 的初衷更吻合。但由于我们几乎是在 llava-v1.5 的 codebase 上做实验，因此就没有过多探究 SigLIP 的解冻与否，直接 follow 了原始 frozen 的设定

**Comment by wangjiyuan9** [comment]
感谢您的回复！请问能否公布一下3中提到的finetune后的模型，我想对其特性进行研究。如果不可以，能否告知一下训练细节，比如image net的具体训练集，还有学习率，解冻等等的设置。
再次感谢您的快速回复！

**Comment by Haochen-Wang409** [comment]
我们没有保存imagenet finetune后的denoise。具体来说，我们使用了bs=256, lr=1e-4，只训练denoiser

**Comment by wangjiyuan9** [comment]
请问具体的split是啥呢？我使用release的ROSS进行finetune，能否保证完美复现论文的效果呢？因为下载imagenet耗时耗力，我还是希望您能提供checkpoint，如果复现不了我可能还要和您多次沟通，谢谢

**Comment by Haochen-Wang409** [comment]
我们是在ImageNet-1K training set 上训的，follow 了 MAR 的 implementation。如果你需求效率，建议自己改一下，或者等我后续空闲重新训一下。

**Comment by wangjiyuan9** [comment]
1.谢谢您的回复，请问MAR具体是指这篇paper吧：
**Autoregressive Image Generation without Vector Quantization**
我比较好奇如果没有文本的话，LM loss是完全不参与监督吗？能否大致说一下代码的修改呢？

> 1. Released 是最终与 sota 比较的 Ross
> 2. 看上去不是特别正常，你是加载了 Ross 的权重然后在 558K 上训练的吗？
> 3. ImageNet finetune 的流程大致是：图 -> SigLIP (frozen) -> Qwen2 (frozen) -> denoiser (trainable) -> predicted noise。关于”但如果没有llm的训练，是否更少的epoch就可以得到好的diffusion model“这一点，我没有特别理解，你能详细解释一下吗？

2.抱歉关于上面的这个问题3我再解释一下，我的意思是，我们现在使用完全初始化的denoiser训练（设置A），和使用ROSS训练后的denoiser模型做起点训练（设置B），在同样的ROSS LLM模型下，能说明ROSS的denoiser训练的更快吗？A是5个epoch，如果B是需要超过5个epoch才能到A的效果吗？

**Comment by Haochen-Wang409** [comment]
因为我们已经在微信取得了交流，因此我将关闭这个 issue
